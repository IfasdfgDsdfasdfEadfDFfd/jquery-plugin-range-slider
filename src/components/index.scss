@import './variables.scss';
@import './input-range.scss';
@import './slider-track.scss';
@import './slider-progress.scss';
@import './slider-thumb.scss';


.range-slider {
  position: relative;
  height: $thumb-size;
  margin-top: 2.5rem;
}

.range-slider--vertical {
  transform: rotate(270deg);
  margin: 40% 15%;

  & .range-slider__thumb__marker {
    transform: rotate(90deg);
  }
}
