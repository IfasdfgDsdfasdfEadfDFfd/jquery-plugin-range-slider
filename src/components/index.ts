import { RangeSlider } from "./range-slider/range-slider.view";
import { createStore, loadFromLocalStoragePlugin, NaNValidator, saveToLocalStoragePlugin } from "../core";
import { IRangeSliderStore, rangeSliderStoreReducer } from "./reducer";


export function createRangeSlider(root: HTMLElement, initStoreValue: Partial<IRangeSliderStore>) {
  const STORE_ID = 'range-sldider-store-id';
  const INIT_VALUE: IRangeSliderStore = Object.assign({
    min: 0,
    max: 100,
    step: 5,
    value: [20, 80],
    vertical: false,
    markerVisibility: false,
    intervalMode: true,
  }, initStoreValue);

  const store = createStore<IRangeSliderStore>(INIT_VALUE, rangeSliderStoreReducer, {
    pre: [loadFromLocalStoragePlugin(STORE_ID)],
    post: [saveToLocalStoragePlugin(STORE_ID)],
  }, [NaNValidator]);

  const rangeSlider = new RangeSlider(store);
  root.appendChild(rangeSlider.root.element);

  store.coldStart();

  return store;
}
