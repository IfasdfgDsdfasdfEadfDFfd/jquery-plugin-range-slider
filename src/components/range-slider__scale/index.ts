export * from './range-slider__scale.view';
export * from './range-slider__scale.model';
export * from './range-slider__scale.controller';
export * from './range-slider__scale.component';
