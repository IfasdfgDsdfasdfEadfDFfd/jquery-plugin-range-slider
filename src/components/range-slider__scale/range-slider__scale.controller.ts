import { Controller } from '@core';

class ScaleController extends Controller {
  mapState({ scale }: RangeSliderModelData): Partial<ScaleProps> {
    return {
      items: scale.values,
    };
  }
}

export { ScaleController };
