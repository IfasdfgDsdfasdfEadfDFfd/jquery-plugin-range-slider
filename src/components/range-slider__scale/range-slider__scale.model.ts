import { Model } from '@core';

class ScaleModel extends Model {
  name = 'scale';

  data = {
    values: [],
  };
}

export { ScaleModel };
