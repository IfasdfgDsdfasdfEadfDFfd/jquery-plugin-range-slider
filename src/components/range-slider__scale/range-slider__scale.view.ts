import { View } from '@core';
import { memo } from 'core/utils';

@memo(['updateItems'])
class Scale extends View<ScaleProps> {
  tag = 'ul';
  attrs = {
    class: 'range-slider__track-scale',
  };

  render({ items }: ScaleProps): void {
    this.updateItems(items);
  }

  updateItems(newItems: string[]): void {
    this.nativeElement.innerHTML = '';

    this.children = newItems.reduce((newChildren, value, index) => {
      const item = new ScaleItem();

      item.init(this.nativeElement);
      item.render({ buttonText: value });

      return { ...newChildren, [index]: item };
    }, {} as ViewChildren);
  }
}

class ScaleItem extends View<ScaleItemProps> {
  tag = 'li';
  attrs = {
    class: 'range-slider__track-scale__item',
  };
  children = {
    button: new ScaleItemButton(),
  };

  render({ buttonText }: ScaleItemProps): void {
    this.children.button.render({ text: buttonText });
  }
}

@memo(['updateText'])
class ScaleItemButton extends View<ScaleItemButtonProps> {
  tag = 'button';
  attrs = {
    type: 'button',
    class: 'range-slider__track-scale__item-button',
  };

  render({ text }: ScaleItemButtonProps): void {
    this.updateText(text);
  }

  updateText(newText: string): void {
    this.nativeElement.textContent = newText;
  }
}

export { Scale, ScaleItem, ScaleItemButton };
