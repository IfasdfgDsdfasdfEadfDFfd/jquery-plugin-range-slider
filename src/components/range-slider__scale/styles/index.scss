.range-slider__scale {
  margin: 0;
}
