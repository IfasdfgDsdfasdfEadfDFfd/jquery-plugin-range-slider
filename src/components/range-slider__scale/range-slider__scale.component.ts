import { Component } from '@core';

import { ScaleController } from './range-slider__scale.controller';
import { ScaleModel } from './range-slider__scale.model';
import { Scale } from './range-slider__scale.view';

class ScaleComponent extends Component {
  view = new Scale();
  model = new ScaleModel();
  controller = new ScaleController();
}

export { ScaleComponent };
