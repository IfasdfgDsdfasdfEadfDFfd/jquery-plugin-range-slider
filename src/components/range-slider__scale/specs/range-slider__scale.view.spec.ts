import { ScaleItemButton } from 'components/range-slider__scale';
import { ScaleItem } from '../range-slider__scale.view';

describe('ScaleItem', () => {
  let item: ScaleItem;
  const buttonText = 'button text';

  beforeEach(() => {
    item = new ScaleItem();
    item.init();
    item.render({ buttonText });
  });

  test('to match snapshot', () => {
    expect(item.nativeElement).toMatchSnapshot();
  });
});

describe('ScaleItemButton view', () => {
  let button: ScaleItemButton;
  const text = 'button text';

  beforeEach(() => {
    button = new ScaleItemButton();
    button.init();
    button.render({ text });
  });

  test('updateText()', () => {
    expect(button.nativeElement.textContent).toEqual(text);

    const newText = 'new button text';
    button.updateText(newText);

    expect(button.nativeElement.textContent).toEqual(newText);
  });

  test('to match snapshot', () => {
    expect(button.nativeElement).toMatchSnapshot();
  });
});
