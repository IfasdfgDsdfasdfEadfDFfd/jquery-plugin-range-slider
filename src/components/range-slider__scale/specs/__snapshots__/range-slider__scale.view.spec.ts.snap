// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale to match snapshot 1`] = `
<ul
  class="range-slider__scale"
>
  <li
    class="range-slider__scale__item"
  >
    <button
      class="range-slider__scale__item-button"
      type="button"
    >
      first
    </button>
  </li>
  <li
    class="range-slider__scale__item"
  >
    <button
      class="range-slider__scale__item-button"
      type="button"
    >
      second
    </button>
  </li>
  <li
    class="range-slider__scale__item"
  >
    <button
      class="range-slider__scale__item-button"
      type="button"
    >
      third
    </button>
  </li>
</ul>
`;

exports[`ScaleItem to match snapshot 1`] = `
<li
  class="range-slider__scale__item"
>
  <button
    class="range-slider__scale__item-button"
    type="button"
  >
    button text
  </button>
</li>
`;

exports[`ScaleItemButton to match snapshot 1`] = `
<button
  class="range-slider__scale__item-button"
  type="button"
>
  button text
</button>
`;
